### PR TITLE
More Lead School Refactoring to Lead Partner

### PR DESCRIPTION
**Pull request by adamjcolvin**
### Context
[Trello](https://trello.com/c/z4eDMVkv)

We need to rename every reference to `lead_school` and its variations to `lead_partner`. This will be a big change, so let's use this branch as a feature branch until we're ready. 

### Changes proposed in this pull request
Rename all `lead_school` references to `lead_partner`, fixing any broken tests and removing code where it makes sense. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by kennyevil — APPROVED** [review]

File: app/components/trainee_about/view.html.erb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 <div class="record-details">
   <%= render RecordDetails::View.new(trainee: @trainee,
       last_updated_event: trainee.timeline.first,
-      show_provider: current_user.system_admin? || current_user.lead_school?,
+      show_provider: current_user.system_admin?,
```
Comment (defong) [inline app/components/trainee_about/view.html.erb:13]: Should this not be changed to lead partner?

File: app/views/guidance/_schools.html.erb
Comment (defong) [inline app/views/guidance/_schools.html.erb:1]: Are we sure?

File: spec/lib/user_with_organisation_context_spec.rb [hunk 2]
```diff
@@ -33,7 +32,7 @@
       it { is_expected.to eq(user.providers.first) }
 
       context "user has a lead school and a provider" do
```
Comment (defong) [inline spec/lib/user_with_organisation_context_spec.rb:34]: Is this test still needed?

File: app/views/guidance/_schools.html.erb
Comment (adamjcolvin) [reply app/views/guidance/_schools.html.erb:1]: Ah I hadn't merged with main for a while, this has been put back now.

File: app/components/trainee_about/view.html.erb (hunk 1 above)
Comment (adamjcolvin) [reply app/components/trainee_about/view.html.erb:13]: I'm honestly not sure, but I've changed it to lead_partner for now.

File: spec/policies/user_policy_spec.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@
 describe UserPolicy do
   subject { described_class }
 
-  let(:lead_school) { create(:school, :lead) }
+  let(:school) { create(:school, :lead) }
```
Comment (tomtrentham) [inline spec/policies/user_policy_spec.rb:8]: I think we might need some thought around the concept of still having lead schools like this. Not for this PR though.

File: spec/policies/user_policy_spec.rb [hunk 4]
```diff
@@ -14,13 +14,7 @@
...
   let(:provider_admin_user_context) { UserWithOrganisationContext.new(user: provider_admin_user, session: {}) }
 
-  let(:lead_school_user) { create(:user, :with_lead_school_organisation, lead_schools: [lead_school]) }
-  let(:lead_school_user_context) { UserWithOrganisationContext.new(user: lead_school_user, session: {}) }
-
-  let(:lead_school_admin_user) { create(:user, :with_lead_school_organisation, system_admin: true, lead_schools: [lead_school]) }
-  let(:lead_school_admin_user_context) { UserWithOrganisationContext.new(user: lead_school_admin_user, session: {}) }
-
-  let(:lead_partner) { create(:lead_partner, :lead_school, school: lead_school) }
+  let(:lead_partner) { create(:lead_partner, :hei, school:) }
```
Comment (tomtrentham) [inline spec/policies/user_policy_spec.rb:17]: Does it make sense having `hei` and `school:`, or could this be 
```ruby
create(:lead_partner, :lead_school)
```

**Review by tomtrentham — APPROVED** [review]
Looks sensible overall.

**Pull request merged**